File: ruby/Mediator/seller.rb
class Seller
  attr_accessor :mediator
  attr_accessor :currency
  attr_accessor :sellingPrice
  def initialize(mediator, currency, sellingPrice)
    @mediator = mediator
    @currency = currency
    @sellingPrice = sellingPrice
    @mediator.addSeller(self)
  end

  def getSellingPrice
    return sellingPrice
  end

  def setSellingPrice(sellingPrice)
    @sellingPrice = sellingPrice
  end

  def isBidAccepted(bidInDollars)
    if (bidInDollars <= sellingPrice)
      print "Seller accepts the bid of USD #{bidInDollars} .\n"
      return true
    else
      print"Seller rejects the bid of USD  #{bidInDollars}.\n"
    return false
    end
  end
end

File: java/Mediator/src/Seller.java


public class Seller {
	
	Mediator mediator;
	String currency;
	int sellingPrice;
	int sellingPriceInDollars;
	
	public Seller(Mediator mediator, String currency, int sellingPrice) {
		super();
		this.mediator = mediator;
		this.currency = currency;
		this.sellingPrice = sellingPrice;
		mediator.addSeller(this); //blank
	}

	public int getSellingPrice() {
		return sellingPrice;
	}

	public void setSellingPrice(int sellingPrice) {
		this.sellingPrice = sellingPrice;
	}

	public boolean isBidAccepted(float bidInDollars) {
		if (bidInDollars <= sellingPrice) {
			System.out.println("Seller accepts the bid of USD " + bidInDollars + ".\n");
			return true;
		} else {
			System.out.println("Seller rejects the bid of USD " + bidInDollars + ".\n");
			return false;
		}
	}
}
